community/kernel/src/main/java/org/neo4j/graphdb/notification/Notification.java

```diff
@@ -0,0 +1,153 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphdb.notification;
+
+/**
+ * Representation for notifications found when executing a query.
+ *
+ * A notification can be visualized in a client pinpointing problems or other information about the query.
+ */
+public final class Notification
+{
+    private final NotificationCode code;
+    private final String title;
+    private final String description;
+    private final NotificationSeverityLevel severity;
+    private final InputPosition position;
+
+    /**
+     * use {@link Builder}
+     */
+    private Notification( NotificationCode code, String title, String description, NotificationSeverityLevel severity, InputPosition position )
+    {
+        this.code = code;
+        this.title = title;
+        this.description = description;
+        this.severity = severity;
+        this.position = position;
+    }
+
+    /**
+     * Returns a notification code for the discovered issue.
+     * @return the notification code
+     */
+    public NotificationCode getCode()
+    {
+        return code;
+    }
+
+    /**
+     * Returns a short summary of the notification.
+     * @return the title of the notification.
+     */
+    public String getTitle()
+    {
+        return title;
+    }
+
+    /**
+     * Returns a longer description of the notification.
+     * @return the description of the notification.
+     */
+    public String getDescription()
+    {
+        return description;
+    }
+
+    /**
+     * Returns the severity level of this notification.
+     * @return the severity level of the notification.
+     */
+    public NotificationSeverityLevel getSeverity()
+    {
+        return severity;
+    }
+
+    /**
+     * The position in the query where this notification points to.
+     * Not all notifications have a unique position to point to and should in
+     * that case return {@link InputPosition#empty}
+     *
+     * @return the position in the query where the issue was found, or
+     * {@link InputPosition#empty} if no position is associated with this notification.
+     */
+    public InputPosition getPosition()
+    {
+        return position;
+    }
+
+    /**
+     * Builds a notification message.
+     */
+    public static class Builder
+    {
+        private final NotificationCode code;
+        private final String title;
+        private final String description;
+
+        private NotificationSeverityLevel severity = NotificationSeverityLevel.INFO;
+        private InputPosition position = InputPosition.empty;
+
+        /**
+         * Constructs a NotificationBuilder.
+         *
+         * @param code see {@link Notification#getCode}
+         * @param title see {@link Notification#getTitle}
+         * @param description see {@link Notification#getDescription}
+         */
+        public Builder( NotificationCode code, String title, String description)
```

Comment:
From looking at the usages, it seems like the `title` and `description` are constant (relative to the `NotificationCode`). It would make sense for those two to be part of the `NotificationCode` as well. That way we would keep it all together in one place, rather than spreading out string constant in multiple places.